include/boost/geometry/formulas/area_formulas.hpp

```diff
@@ -0,0 +1,521 @@
+// Boost.Geometry
+
+// Copyright (c) 2015-2016 Oracle and/or its affiliates.
+
+// Contributed and/or modified by Vissarion Fysikopoulos, on behalf of Oracle
+
+// Use, modification and distribution is subject to the Boost Software License,
+// Version 1.0. (See accompanying file LICENSE_1_0.txt or copy at
+// http://www.boost.org/LICENSE_1_0.txt)
+
+#ifndef BOOST_GEOMETRY_FORMULAS_AREA_FORMULAS_HPP
+#define BOOST_GEOMETRY_FORMULAS_AREA_FORMULAS_HPP
+
+#include <boost/geometry/algorithms/detail/flattening.hpp>
+
+namespace boost { namespace geometry { namespace formula
+{
+
+/*!
+\brief Formulas for computing spherical and ellipsoidal polygon area.
+ The current class computes the area of the trapezoid defined by a segment
+ the two meridians passing by the endpoints and the equator.
+\author See
+- Danielsen JS, The area under the geodesic. Surv Rev 30(232):
+61–66, 1989
+- Charles F.F Karney, Algorithms for geodesics, 2011
+https://arxiv.org/pdf/1109.4448.pdf
+*/
+
+template <
+        typename CT,
+        const std::size_t SeriesOrder = 2,
+        bool ExpandEpsN = true
+>
+class area_formulas
+{
+
+public:
+
+    //TODO: move the following to a more general space to be used by other
+    //      classes as well
+    /*
+        Evaluate the polynomial in x using Horner's method.
+    */
+    template <typename NT, typename IteratorType>
+    static inline NT horner_evaluate(NT x,
+                                     IteratorType begin,
+                                     IteratorType end)
+    {
+        NT result(0);
+        IteratorType it = end;
+        do{
+            result = result * x + *--it;
+        }while(it != begin);
+        return result;
+    }
+
+    /*
+        Clenshaw algorithm for summing trigonometric series
+        https://en.wikipedia.org/wiki/Clenshaw_algorithm
+    */
+    template <typename NT, typename IteratorType>
+    static inline NT clenshaw_sum(NT cosx,
+                                  IteratorType begin,
+                                  IteratorType end)
+    {
+        IteratorType it = end;
+        bool odd(true);
+        CT b_k, b_k1(0), b_k2(0);
+        do{
+            CT c_k = odd ? *--it : NT(0);
+            b_k = c_k + NT(2) * cosx * b_k1 - b_k2;
+            b_k2 = b_k1;
+            b_k1 = b_k;
+            odd = !odd;
+        }while(it != begin);
+
+        return *begin + b_k1 * cosx - b_k2;
+    }
+
+    template<typename T>
+    static inline void normalize(T& x, T& y)
+    {
+        T h = hypot(x, y);
```

Comment:
Is this a call of std function hypot? If that's the case then it cannot be used like this since it was introduced in C++11. You should add your own implementation in `utils/math.hpp` falling back to `std::hypot()` in C++11. An alternative would be Boost.Math special function hypot().